backend/game/player/test_player.py

```diff
@@ -93,3 +97,13 @@ def test_player_energy_cannot_go_below_zero(player):
 def test_player_can_not_have_less_than_zero_victory_points(player):
     player.update_victory_points_by(-1)
     assert player.victory_points == 0
+
+
+def test_player_health_cannot_exceed_decreased_max_health(player):
+    player.update_max_health_by(-1)
+    assert player.current_health == player.maximum_health
+
+
+def test_player_health_less_than_increased_max_health(player):
+    player.update_max_health_by(1)
+    assert player.current_health < player.maximum_health
```

Comment:
Should we add tests for the player keep card methods?

Reply:
added tests